### DISCUSSION TITLE
Tag filtering for ignoring articles

### DISCUSSION THREAD

**Discussion by Redsandro** [FreshRSS/FreshRSS#6844] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

I’m always frustrated when I need to dig through articles I don't care about, for example sports in a news feed.

**Describe the solution you’d like**

I would like a simple to use method for ignoring or removing or marking as "read" such articles using tags that news sites often apply to the html article:

```html
<meta content="sports, football, game" property="og:article:tag">
```

In this case, I would like to trash or mark as read **if og:article:tag contains sports**.

**Describe alternatives you’ve considered**
I haven't figured out a way to do this within FreshRSS, but I tried to write a selector in the developer console:

```javascript
document.querySelector('meta[property="og:article:tag"]').getAttribute('content').includes('sports')
# true
```

```javascript
['sports', 'football', 'politics'].some(tag => 
    document.querySelector('meta[property="og:article:tag"]')?.getAttribute('content')?.includes(tag)
)
# true
```

**Additional context**

**Comment by math-GH** [comment, marked as answer]
Have you tried this feature already?
It is in the feed configs.

![grafik](https://github.com/user-attachments/assets/b17821b8-8373-4171-8bef-59f21091129f)

**Comment by Redsandro** [comment]
@math-GH Now that you mention it, I do remember struggling with this some time ago, and I thought it was too limited. It would make my day if you can tell me I'm wrong. 

I thought it only works with predefined google-like searches such as `intitle` and I don't really understand the [documentation](https://freshrss.github.io/FreshRSS/en/users/10_filter.html#with-the-search-field): What does it search? Only predefined filters? Literal meta tags? The graph? 

How do you use a colon in the meta-tag if the meta-tag goes before the colon, i.e. `og:article:tag:sports`? 

Does #tag only search `<meta name="keywords"` or also `<meta property="og:article:tag"`? ~~Also, space implies `AND`, but many tags I want to filter contain spaces. Such as `#FC Twente`.~~

I couldn't get it to work. Is it case sensitive? Do I need to try again? It's difficult to test matches, because you need to wait until a matching article is published. It would be helpful if there was a match test, similar to "Advanced" -> "Article CSS selector on original website" -> eyeball.

**Update:**

> by tag: #tag **or #tag+with+whitespace or #'tag with whitespace'**

I will set some simple tags with both small case and Upper case, and see if they match OpenGraph tags as well as meta keywords. It will take some time before I know if it works. Thanks for reminding me.

**Comment by Alkarex** [comment]
> What does it search?

The built-in feature to automatically mark as read articles based on their tags applies to the tags as identified by FreshRSS and as visible in the UI:

![image](https://github.com/user-attachments/assets/51854321-ad90-4cb0-a5b1-78edeb6fb5dd)

Article tags are taken from the RSS/Atom feed, not from the HTML page.

> I couldn't get it to work.

If you could not make it work, provide and example of feed and corresponding tag searches you have attempted.

> Is it case sensitive?

The normal tag search syntax `#tag` or `#tag+with+whitespace` or `#'tag with whitespace'` is case-insensitive, but it is possible to make it case-sensitive using our [new regex search](https://freshrss.github.io/FreshRSS/en/users/10_filter.html#regex) like `#/^Hello World$/m` (note that the regex mode is especially powerful for tags).

> It's difficult to test matches, because you need to wait until a matching article is published.

To verify that your automatic mark-as-read filter works, you can start by using it to search in the corresponding feed and see what matches. Alternatively, to test your filter for a feed for which you are not interested in keeping the old articles, you can also *Delete all articles* and then *Update feed*:

![image](https://github.com/user-attachments/assets/39fcc924-655c-4a22-a067-3c26c3fd9352)

**Reply by Maverynthia** [reply]
Found this when searching how to ignore/not download/mark as read tags. It doesn't work for me:

Example RSS:
https://www.romhacking.net/romhackingdotnet.rss

I want to NOT download, Mark as read, ignore, send into the sun articles with the tag #ROM+Hacks+News
What I have:
"Mark an article as read": `tag:#ROM+Hacks+News`
"Conditions for content retrieval": `tag: #Translations+News`
`-tag: #ROM+Hacks+News`

YES retrieve content for "Transations News"
NO do NOT retrieve content for "ROM Hacks News"

These are the tags that are on the articles themselves in FreshRSS

Yes all articles are deleted and then updated

<img width="879" height="583" alt="image" src="https://github.com/user-attachments/assets/557ade3a-d18a-4b6e-a1a3-de2302138b1b" />

However ALL articles are marked as unread.

**Reply by Alkarex** [reply]
@Maverynthia The syntax is wrong. Check https://freshrss.github.io/FreshRSS/en/users/10_filter.html#with-the-search-field and higher up here.
Try for instance `#ROM+Hacks+News`

**Reply by Maverynthia** [reply]
I swear I tried just `#tag+word` and it didn't work, but it seems to be working now. Thanks
